src/main/java/graphql/introspection/IntrospectionQuery.java

```diff
@@ -1,109 +1,203 @@
 package graphql.introspection;
 
+import com.google.common.collect.ImmutableList;
 import graphql.PublicApi;
+import graphql.language.Argument;
+import graphql.language.AstPrinter;
+import graphql.language.BooleanValue;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.FragmentDefinition;
+import graphql.language.FragmentSpread;
+import graphql.language.OperationDefinition;
+import graphql.language.SelectionSet;
+import graphql.language.TypeName;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 @PublicApi
-public interface IntrospectionQuery {
+public interface IntrospectionQuery { // todo iwds support
+    static <T> List<T> filter(T... args) {
+        return Arrays.stream(args).filter(Objects::nonNull).collect(Collectors.toList());
+    }
+
+    static String getIntrospectionQuery(
+        boolean descriptions,
+        boolean specifiedByUrl,
+        boolean directiveIsRepeatable,
+        boolean schemaDescription,
+        boolean inputValueDeprecation,
+        int typeRefFragmentDepth
+    ) {
```

Comment:
A 6 parameter method is a terrible interface for generating introspection queries

```
getIntrospectionQuery(true, false, true, false, true, 7);
```

Looking at that code its near impossible to get signal on what you are getting - and Java doesnt have named parameters which makes it worse

I think this needs a builder object (aka options).  See `graphql.schema.idl.SchemaPrinter.Options` for inspiration

I also think this should go into a class (not the interface) 

Hence this might be

```
String INTROSPECTION_QUERY = IntrospectionQueryBuilder.build(IntrospectionQueryBuilder.defaultOptions())

Reply:
a great example of how IntelliJ has spoiled me, I can surely improve this with a builder.